tasks/push-snapshot/push-snapshot.yaml

```diff
@@ -82,79 +101,62 @@ spec:
+              exit 1
             fi
-            if [[ $(jq -r ".images.addGitShaTag" "${DATA_FILE}") != "false" ]] ; then # Default to true
-              if [ "${git_sha}" != "null" ] ; then
-                push_image "${name}" "${containerImage}" "${repository}" "${git_sha:0:7}"
-                push_image "${name}" "${containerImage}" "${repository}" "${git_sha}"
-              else
-                printf 'Asked to create git sha based tag, but no git sha found in %s\n' "${component}"
-                exit 1
-              fi
+          fi
+          # Push the associated source container using the common tag
+          if [[ $(jq -r ".images.pushSourceContainer" "${DATA_FILE}") == "true" ]] ; then # Default to false
+            # Calculate the source container image based on the provided container image
+            sourceContainer="${containerImage%@sha256:*}:${git_sha}.src"
+            # Check if the source container exists
+            if ! skopeo inspect --no-tags "docker://${sourceContainer}" &>/dev/null; then
+              echo "Error: Source container ${sourceContainer} not found!"
+              exit 1
             fi
-            if [[ $(jq -r ".images.addSourceShaTag" "${DATA_FILE}") != "false" ]] ; then # Default to true
-              if [[ "${containerImage}" == *"@sha256"* && $(echo "${containerImage}" | tr -cd ':' | wc -c) -eq 1 ]]
-              then
-                sha=$(echo "${containerImage}" | cut -d ':' -f 2)
-                push_image "${name}" "${containerImage}" "${repository}" "${sha}"
-              else
-                printf 'Asked to create source sha based tag, but no sha found in %s\n' "${containerImage}"
-                exit 1
-              fi
+            if [ $floatingTagsCount -eq 0 ]; then
+              echo "Error: at least one tag must exist in floatingTags when pushing source containers"
+              exit 1
             fi
-            # Push the associated source container using the common tag
-            if [[ $(jq -r ".images.pushSourceContainer" "${DATA_FILE}") == "true" ]] ; then # Default to false
-              # Calculate the source container image based on the provided container image
-              sourceContainer="${containerImage%@sha256:*}:${git_sha}.src"
-              # Check if the source container exists
-              if ! skopeo inspect --no-tags "docker://${sourceContainer}" &>/dev/null; then
-                echo "Error: Source container ${sourceContainer} not found!"
-                exit 1
-              fi
-              if [ -z "$floatingTag" ]; then
-                echo "Error: floatingTag needs to be set when pushing source containers"
-                exit 1
-              fi
+            for floatingTag in $(jq -rc '.images.floatingTags[]' $DATA_FILE) ; do
               push_image "${name}" "${sourceContainer}" "${repository}" "${floatingTag}-${timestamp}-source"
```

Comment:
I wonder if we should also push `"${floatingTag}-source". It seems weird that for binary images we will push both with and without timestamp, but for source, we only push with timestamp. I think we should ask somebody for clarification here. @ralphbean ?

Reply:
I'd like to add that I don't think this needs to be resolved here and now. So feel free to proceed with merging if you like. This can always be changed later.

Reply:
Cool, if CI passes before Ralph answers, I'll just merge and open a new PR based on his response

Reply:
Yeah, tbh, it's not a requirement from the end user (legal) to use "${floatingTag}-source" but, I like the symmetry of it.  +1 from me to add that, but no need to hold up this change without it.

Reply:
Updated to also push `${floatingTag}-source`

Reply:
Awesome, thanks to both of you!